### PR TITLE
Involutive fix

### PR DESCRIPTION
**Pull request by tonyday567**
Added tests and polymorphed Complex instances

### CODE REVIEW

**Comment by tonyday567** [comment]
@ocramz can you have a quick look at the laws and changes?  Marked two hacky bits.

Also is `==>` a widely used convention?  Should I change all the law comments to this?

File: numhask-prelude/test/test.hs [hunk 1]
```diff
@@ -93,3 +94,12 @@ testsComplexFloat =
...
       testLawOf ([] :: [Complex Float]) <$> metricComplexFloatLaws
     ]
+
+testsComplexInvolutive :: TestTree
+testsComplexInvolutive =
+  testGroup
+    "Complex Involutive (Float)"
+    [ testGroup "Involutive Ring" $
+      -- fixme: why doesn't Complex Int work here?
+      testLawOf ([] :: [Complex Float]) <$> involutiveRingLaws
```
[the remark this answers is not in the document]
Comment (ocramz) [reply numhask-prelude/test/test.hs]: Some instances of the base typeclasses have a RealFloat constraint but that shouldn't apply here :/ https://hackage.haskell.org/package/base-4.11.0.0/docs/Data-Complex.html I'm puzzled as you are for now

File: numhask-prelude/src/NumHask/Laws.hs [hunk 2]
```diff
@@ -541,3 +542,15 @@ starSemiringLaws = semiringLaws <>
...
 
+-- | involutive ring
+involutiveRingLaws :: (Eq a, InvolutiveRing a) => [Law a]
+involutiveRingLaws =
+    [ ( "adjoint plus law: adj (a + b) ==> adj a + adj b"
+    , Binary (\a b -> adj (a `plus` b) == adj a `plus` adj b))
+    , ( "adjoint times law: adj (a * b) ==> adj b * adj a"
+    , Binary (\a b -> adj (a `times` b) == adj b `times` adj a))
+    , ( "adjoint multiplicative unit law: adj one ==> one"
+    , Unary (\a -> adj one + a == one + a)) -- fixme: crazy way to get the type
```
[the remark this answers is not in the document]
Comment (ocramz) [reply numhask-prelude/src/NumHask/Laws.hs]: How do you mean?

**Comment by ocramz** [comment]
re. `==>` : Oh, I thought it was a DocTest thing!

File: numhask-prelude/src/NumHask/Laws.hs (hunk 2 above)
[the remark this answers is not in the document]
Comment (tonyday567) [reply numhask-prelude/src/NumHask/Laws.hs]: `(adj one == one)` is an ambiguous type.  So I need to annotate, but `(adj one = (one :: a))` doesn't work.  The only thing I can think of is monomorph the type sig...

[the remark this answers is not in the document]
Comment (ocramz) [reply numhask-prelude/src/NumHask/Laws.hs]: Would throwing in a `forall a.` help ?

**Pull request merged**